distribution/lib/Standard/Database/0.0.0-dev/src/Internal/SQLite/SQLite_Dialect.enso

```diff
@@ -74,11 +79,30 @@ type SQLite_Dialect
         False ->
             Order_Descriptor.Value internal_column.expression sort_direction collation=Nothing
 
+    ## PRIVATE
+       Prepares a join operation, returning a new table instance encapsulating a
+       proper query.
+    prepare_join : Connection -> Join_Kind -> Text -> From_Spec -> From_Spec -> Vector -> Vector -> Vector -> Table
+    prepare_join self connection join_kind new_table_name left_subquery right_subquery on_expressions where_expressions columns_to_select = case join_kind of
+        Join_Kind.Right ->
+            # We just do a left join with swapped order of sub-queries, while keeping the original order of columns.
+            Join_Helper.default_prepare_join connection Join_Kind.Left new_table_name right_subquery left_subquery on_expressions where_expressions columns_to_select
+        Join_Kind.Full ->
+            # TODO workaround for full joins by left outer + right-anti union
+            Error.throw (Unsupported_Database_Operation.Error "Full outer joins are not YET supported by the SQLite backend. You may need to materialize the Table to perform this operation.")
+        _ ->
+            # Other kinds of joins just fall back to the default logic.
+            Join_Helper.default_prepare_join connection join_kind new_table_name left_subquery right_subquery on_expressions where_expressions columns_to_select
+
 ## PRIVATE
 make_internal_generator_dialect =
     text = [starts_with, contains, ends_with]+concat_ops
     counts = [agg_count_is_null, agg_count_empty, agg_count_not_empty, ["COUNT_DISTINCT", agg_count_distinct], ["COUNT_DISTINCT_INCLUDE_NULL", agg_count_distinct_include_null]]
     stats = [agg_stddev_pop, agg_stddev_samp]
+
+    # TODO these may be problematic as they may break indexes, double-check query plans!
```

Comment:
is this still pending?

Reply:
Yes, we need further discussion on this. It will be done in a separate task.